apps/self-hosted/hosting/scripts/hivesigner-redirect-uris.py

```
@@ -131,6 +287,171 @@ def write_private(contents: str, path: str | None) -> str:
+    handle back made staying locked the caller's job, and the caller dropped it on
+    the floor: the lock was taken and released before the account was even read,
+    and the whole thing looked like it was working. There is now no return value
+    to discard, and `test_the_lock_outlives_the_call` fails if this stops holding.
+    """
+    global _LOCK_HANDLE
+    handle = open(LOCK_FILE, "w", encoding="utf-8")
+    try:
+        fcntl.flock(handle, fcntl.LOCK_EX | fcntl.LOCK_NB)
+    except BlockingIOError:
+        handle.close()
+        raise SystemExit(0)
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)